### PR TITLE
Sec-Fetch-Site's scheme comparison should account for opaque origins.

### LINKED ISSUE #41: Comparison does not seem to account for opaque origins
At least, the text I read in 43165833dbb66890aa3739161dfc9c00dc0db09c seems to assume the origin has a scheme, which isn't always a given.

### PR DESCRIPTION
**Pull request by mikewest**
Thanks, @annevk. Closes #41.

### CODE REVIEW

**Comment by mikewest** [comment]
WDYT, @annevk?

File: index.bs [hunk 1]
```diff
@@ -248,10 +248,11 @@ To <dfn abstract-op lt="set-site">set the `Sec-Fetch-Site` header</dfn> for a [=
...
         1.  If |url| is [=same origin=] with |r|'s [=request/origin=], [=iteration/continue=].
 
-        2.  If |r|'s [=request/origin=]'s [=origin/scheme=] is not the same as |url|'s
-            [=url/scheme=], or if |r|'s [=request/origin=]'s [=registrable domain=] is not the same
-            as |url|'s [=registrable domain=], set |header|'s value to `cross-site` and
-            [=iteration/break=].
+        2.  Set |header|'s value to `cross-site` and break if any of the following are true:
+
+            *  |r|'s [=request/origin=] is an [=opaque origin=].
+            *  |r|'s [=request/origin=]'s [=origin/scheme=] is not the same as |url|'s
```
Comment (annevk) [inline index.bs]: need to add url/scheme at the end here?

Comment (mikewest) [reply index.bs]: Yup, that was dumb. Fixed in https://github.com/w3c/webappsec-fetch-metadata/pull/42/commits/f3cd5c10ff4c627e2bc8de4447a59a9ac7b2bd1e.

**Comment by annevk** [comment]
Seems better, though dot usage is inconsistent.

I think you still want to use https://url.spec.whatwg.org/#host-same-site as there's a number of cases this doesn't consider, such as IP addresses.

**Comment by mikewest** [comment]
Right on both counts. Thanks!

**Pull request merged**